mypy/test/data/check-statements.test

```diff
@@ -51,6 +51,27 @@ def g() -> None:
 [out]
 main: note: In function "g":
 
+[case testReturnInGenerator]
+from typing import Generator
+def f() -> Generator[int, None, str]:
+    yield 1
+    return "foo"
+[out]
+
+[case testEmptyReturnInGenerator]
+from typing import Generator
+def f() -> Generator[int, None, str]:
+    yield 1
+    return  # E: Return value expected
+[out]
+main: note: In function "f":
+
+[case testReturnInIterator]
+from typing import Iterator
+def f() -> Iterator[int]:
+    yield 1
+    return "foo"
+[out]
```

Comment:
I guess there are also cases where the return type is `Generator[..., ... None]` and there's a plain `return` in the body (no error) or a `return 42` (error)?